app/controllers/devise_token_auth/concerns/set_user_by_token.rb

```diff
@@ -68,17 +68,17 @@ def set_user_by_token(mapping=nil)
     return @resource if @resource && @resource.is_a?(rc)
 
     # ensure we clear the client_id
-    if !@token
-      @client_id = nil
+    if !@token.token
```

Comment:
Calling `@token.token` seems like a 'smell'... wouldn't it be more idiomatic ruby to say something like `@token` or `@token.present?`?

Reply:
You are right.
`@token.present?` looks good.
But what should be checked of token object? client_id, token_hash, token, or combinations of them, or all together?
Could it wait for further pull requests? With `TokenFactory` implementation it will be easy (I hope).

Reply:
I think you should define the `present?` method with only the token so we don't change the behavior.